iot-hub/Quickstarts/simulated-device-2/Program.cs

```diff
@@ -0,0 +1,147 @@
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// This application uses the Azure IoT Hub device SDK for .NET
+// For samples see: https://github.com/Azure/azure-iot-sdk-csharp/tree/master/iothub/device/samples
+
+using Microsoft.Azure.Devices.Client;
+using System;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SimulatedDevice
+{
+    /// <summary>
+    /// This sample illustrates the very basics of a device app sending telemetry and receiving a command.
+    /// For a more comprehensive device app sample, please see
+    /// <see href="https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/master/iot-hub/Samples/device/DeviceReconnectionSample"/>.
+    /// </summary>
+    internal class Program
+    {
+        private static DeviceClient s_deviceClient;
+
+        // The device connection string to authenticate the device with your IoT hub.
+        // Using the Azure CLI:
+        // az iot hub device-identity show-connection-string --hub-name {YourIoTHubName} --device-id MyDotnetDevice --output table
+        private static string s_connectionString = "{Your device connection string here}";
+
+        private static TimeSpan s_telemetryInterval = TimeSpan.FromSeconds(1); // Seconds
+
+        private static async Task Main(string[] args)
+        {
+            Console.WriteLine("IoT Hub Quickstarts #1 - Simulated device.");
+
+            // This sample accepts the device connection string as a parameter, if present
+            CheckForConnectionStringArgument(args);
+
+            // Connect to the IoT hub using the MQTT protocol
+            s_deviceClient = DeviceClient.CreateFromConnectionString(s_connectionString, TransportType.Mqtt);
+
+            // Create a handler for the direct method call
+            await s_deviceClient.SetMethodHandlerAsync("SetTelemetryInterval", SetTelemetryInterval, null);
+
+            // Set up a condition to quit the sample
+            Console.WriteLine("Press control-C to exit.");
+            using var cts = new CancellationTokenSource();
+            Console.CancelKeyPress += (sender, eventArgs) =>
+            {
+                eventArgs.Cancel = true;
+                cts.Cancel();
+                Console.WriteLine("Device simulator exit requested...");
+            };
+
+            // Run the telemetry loop
+            await SendDeviceToCloudMessagesAsync(cts.Token);
+
+            Console.WriteLine("Device simulator finished.");
+        }
+
+        private static void CheckForConnectionStringArgument(string[] args)
+        {
+            if (args.Any())
+            {
+                try
+                {
+                    var cs = IotHubConnectionStringBuilder.Create(args.First());
+                    s_connectionString = cs.ToString();
+                }
+                catch (Exception) { }
```

Comment:
if we are not doing anything in the catch, we probably don't need the try block either.
I personally think that if this throws we should exit with an error.

Reply:
The parameter is optional, and I only want to set it if it passes parsing.